tests/test_operation.py

```diff
@@ -2826,6 +2827,57 @@ class CustomOperator(qml.operation.Operator):
     assert qml.equal(new_op, CustomOperator(2.3, wires=0))
 
 
+@pytest.mark.parametrize(
+    "coeffs, obs",
+    [
+        (
+            [1.5, 0.5, 1, 1],
+            [qml.Identity(1), qml.Z(1) @ qml.Z(2), qml.X(1) @ qml.Y(2), qml.Hadamard(1)],
+        ),
+        ([0.5], [qml.X(1)]),
+        ([1], [qml.X(0) @ qml.Y(1)]),
```

Comment:
Do you want these to be `Tensor` or `Prod` instances? I would make it explicit (i.e. `qml.prod(X(0), Y(1))` or `Tensor(X(0), Y(1))` in the tests